### PR TITLE
Hard-code versions used in History upgrade/downgrade conversions.

### PR DESCRIPTION
**Pull request by ceastlund**
Re-creating this PR after making tests for history conversions (see `astlib/test/test_history.ml`). The tests forced me to fix prior issues, such as getting `src` and `dst` versions the wrong way around in the hard-coded conversions.

### CODE REVIEW

File: astlib/history.mli [hunk 1]
```diff
@@ -27,8 +27,8 @@ val convert
 
 type 'node conversion_function
   = 'node Ast.t
-  -> to_ast:('node -> version:Version.t -> 'node Ast.t)
-  -> of_ast:('node Ast.t -> version:Version.t -> 'node)
+  -> to_src_ast:('node -> 'node Ast.t)
```
Comment (NathanReb) [inline astlib/history.mli]: I'm not sure what the best terminology is here but I find those name a bit confusing.

What helped me writing the migration functions was seeing them as a deconstructor and a builder of some sort. In other words, you use `to_src_ast` to deconstruct `src` values and pattern match over them and use `of_dst_ast` to build new `dst` values. What would you think about reusing that terminology here?

**Review by NathanReb — APPROVED** [review]
It looks good! I have a minor comment about naming but the PR is already an improvement on its own so we can improve those names later!

File: astlib/history.mli (hunk 1 above)
Comment (NathanReb) [reply astlib/history.mli]: I think overall it's okay because the types won't let you misuse those functions anyway.

I just find it particularly hard because the `to`/`of` prefixes are swapped compared to my mental model of what those functions do.

Comment (ceastlund) [reply astlib/history.mli]: I pushed another commit naming things as `wrap` / `unwrap`, I think it comes out much better. In the upgrade/downgrade implementation they're a bit wordy, `unwrap_src_version` and `wrap_dst_version`,  I erred on the side of clarity. What do you think?

Comment (ceastlund) [reply astlib/history.mli]: Or should those just be `wrap` and `unwrap` as well, with the versions being implied?

Comment (ceastlund) [reply astlib/history.mli]: Never mind, I went ahead and made them all `wrap` and `unwrap`.
